ethereum-consensus/src/deneb/blob_sidecar.rs

```diff
@@ -27,6 +27,28 @@ pub struct BlobSidecar<
     pub kzg_commitment_inclusion_proof: Vector<Bytes32, KZG_COMMITMENT_INCLUSION_PROOF_DEPTH>,
 }
 
+impl<const BYTES_PER_BLOB: usize, const KZG_COMMITMENT_INCLUSION_PROOF_DEPTH: usize>
+    BlobSidecar<BYTES_PER_BLOB, KZG_COMMITMENT_INCLUSION_PROOF_DEPTH>
+{
+    pub fn verify_blob_sidecar_inclusion_proof(&mut self) -> bool {
+        // TODO: Calculate real gindex
+        let gindex = 4;
+
+        // TODO: Handle hash_tree_root() error
+        let leaf = &self.kzg_commitment.hash_tree_root().unwrap();
+        let branch: Vec<_> = self
+            .kzg_commitment_inclusion_proof
+            .iter()
+            .map(|bytes| Node::try_from(bytes.as_slice()).unwrap())
+            .collect();
+        let depth = KZG_COMMITMENT_INCLUSION_PROOF_DEPTH;
+        let index = gindex;
+        let root = &self.signed_block_header.message.body_root;
+
+        is_valid_merkle_branch(leaf, branch.iter(), depth, index, root)
```

Comment:
I actually think I'll update this API some and include a bespoke error type, rather than just `bool`

we can make the changes once I land this and the g-index updates to `ssz-rs`